phoenix/app/cad_products_phoenix/lib/cad_products_phoenix_web/services/product.ex

```diff
@@ -0,0 +1,51 @@
+defmodule CadProductsPhoenixWeb.Services.Product do
+
+  alias CadProductsPhoenix.Cache
+  alias CadProductsPhoenix.Management
+  alias CadProductsPhoenix.ProductIndex
+
+  def fetch_all() do
+    case Cache.get("products") do
+      {:ok, _} = result ->
+        ProductIndex.index_products()
```

Comment:
Dessa forma vai pesar muito tua aplicação, sempre que o cliente acessar o index terá que buscar todos os produtos do banco e indexa-los 1 a 1 no ELS. Uma forma melhor seria: indexar o produto quando for criado/atualizado e removê-lo do indíce quando for deletado.